pydatastructs/trees/m_ary_trees.py

```diff
@@ -77,7 +77,13 @@ def insert(self, key, data):
 
         None
         """
-        raise NotImplementedError("This is an abstract method.")
+        temp = self.search(key)
+        if temp.is_root is True and temp.key is None:
+            temp.key = key
+            temp.data = data
+        if temp is not None:
+            self.tree[temp].data = data
+
```

Comment:
Please see https://github.com/codezonediitj/pydatastructs/issues/101#issuecomment-595095244

Reply:
So we need only search function for m-ary?

Reply:
`MAryTree` is abstract. `MArySearchTree` is concrete in the way insertion and search is done, so both of these method will be defined in that class.